gp2-stress-experiment.ipynb

```diff
@@ -0,0 +1,903 @@
+    "# maybe another masking before summation, with tougher threshold to exclude boderline pixels.\n",
+    "sample_final = weighted_sum(rebinned_wavelength[\"sample\"], \"spectrum\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "outputs": [],
+   "source": [
+    "rebinned_wavelength.coords[\"detector_mapping\"] = imaging.make_detector_groups(bank_width, bank_width,\n",
+    "                                                                              nx_target, ny_target)\n",
+    "\n",
+    "grouped = sc.groupby(rebinned_wavelength, \"detector_mapping\").sum(\"spectrum\")\n",
+    "grouped_sample_elastic = grouped[\"sample_elastic\"]"
+   ],
+   "metadata": {
+    "collapsed": false,
+    "pycharm": {
+     "name": "#%%\n"
+    }
+   }
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "pycharm": {
+     "name": "#%%\n"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "histogram_matrices = [sample_final, grouped_sample_elastic]\n",
+    "template_histogram = histogram_matrices[1]\n",
+    "\n",
+    "# Average out detector positions\n",
+    "positions_dataset = sc.Dataset()\n",
+    "positions_dataset[\"position\"] = sc.Variable(rebinned_wavelength.coords[\"position\"])\n",
+    "positions_dataset.coords[\"detector_mapping\"] = rebinned_wavelength.coords[\"detector_mapping\"]\n",
+    "grouped_positions = sc.groupby(positions_dataset, \"detector_mapping\").mean(\"spectrum\")\n",
+    "\n",
+    "plots = []\n",
+    "for spec_num in template_histogram.coords[\"detector_mapping\"].values:\n",
+    "    pos = grouped_positions[\"position\"][\"detector_mapping\", int(spec_num)]\n",
+    "    # Append spectrum index and position\n",
+    "    plots.append((spec_num,pos))\n",
+    "\n",
+    "# We cant access position using (dim, I), since the vector 3d has not dim currently\n",
+    "Y_key = 1\n",
+    "X_key = 0\n",
+    "\n",
+    "# We must use round to prevent floating points errors during sorting like -0.05 > 0.05\n",
+    "plots.sort(key=lambda t: (round(-t[1].value[Y_key], 10), round(t[1].value[X_key], 10)))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "pycharm": {
+     "name": "#%%\n"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "# Bragg edge position roughly, in Angstrom\n",
+    "FCC_a = 3.5 # Aangstrom, taken from COD entry 9008469\n",
+    "BCC_a = 2.865 # Angstrom, average value from COD\n",
+    "indices_FCC = [(1,1,1),(2,0,0),(2,2,0),(3,1,1)]#,(2,2,2)]#,(4,0,0)]\n",
+    "indices_BCC = [(1,1,0),(2,0,0),(2,1,1),(2,2,0)]#,(3,1,0),(2,2,2)]\n",
+    "\n",
+    "def create_Braggedge_list(lattice_constant, Miller_indices):\n",
+    "    '''\n",
+    "    :param Miller_indices: like [(1,1,0),(2,0,0),...]\n",
+    "    :type Miller_indices: list of tuples\n",
+    "    '''\n",
+    "    from numpy import sqrt\n",
```

Comment:
Can omit this. Already import numpy as np, just np.sqrt dirctly, may be other places that do this.